libr/io/p/io_ptrace.c

```diff
@@ -43,8 +43,13 @@ static int __waitpid(int pid) {
 	return (waitpid (pid, &st, 0) != -1);
 }
 
+#if defined(__GLIBC__) && defined(__linux__)
 #define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), NULL)
 #define debug_write_raw(io,x,y,z) r_io_ptrace((io), PTRACE_POKEDATA, (x), (void *)(y), (void *)(z))
+#else
+#define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), 0)
```

Comment:
is there no `NULL` on BSD?

Reply:
There is but as you pointed out there is r_ptrace_data_t and they have different underlying types Linux and non Linux for a reason.

Reply:
I see. Then you could use the already existing `#if defined...` in r_io.h and do something like:
```C
#if defined(__GLIBC__) && defined(__linux__)
typedef enum __ptrace_request r_ptrace_request_t;
typedef void * r_ptrace_data_t;
#define R_PTRACE_NODATA NULL
#else
typedef int r_ptrace_request_t;
typedef int r_ptrace_data_t;
#define R_PTRACE_NODATA 0
#endif
```
```C
#define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), R_PTRACE_NODATA)
```
and avoid all together the new `#if defined...` you are introducing in this PR. This way things will be more "declarative" and there will be less `#if..` to follow.

Reply:
Ok did a PR for this little code simplification.

Reply:
(ops I didn't see the PR was already merged...)

Reply:
That s ok there is no functional change nor bug introduction ;-)